src/test/java/com/wnis/linkyway/utils/ResponseBodyMatchers.java

```diff
@@ -0,0 +1,74 @@
+package com.wnis.linkyway.utils;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException;
+import org.springframework.test.web.servlet.ResultMatcher;
+
+import java.lang.reflect.Field;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ *
+ * ResponseBodyMatcher is very useful when using mvcMock.
+ *
+ * <p>This can be used to verify the contents of the body for the response when testing the controller.</p>
+ *<p>example</p>
+ * <pre> {@code
+ *                 MvcResult mvcResult = mockMvc.perform(post("/api/tags")
+ *                         .contentType("application/json")
+ *                         .content(objectMapper.writeValueAsString(tagRequest)))
+ *                         .andExpect(status().is(400))
+ *                         .andExpect(responseBody() // create ResponseBodyMatcher
+ *                                 .containsPropertiesAsJson(ErrorResponse.class)) // method
+ *                         .andReturn();
+ *
+ * }
+ *
+ * </pre>
+ *
+ * @author Hyeongyu
+ * @version 1.0.0
+ */
+public class ResponseBodyMatchers {
+    private ObjectMapper objectMapper = new ObjectMapper();
+    /**
+     * <p>Validate that the field name and value of the ResponseBody object are the same as expected</p>
+     * <p>you do not need to create Object for test. just use *.class</p>
+     * @param expectedObject Object created for test
+     * @param targetClass Expected class structure (*.class)
+     */
+    public <T>ResultMatcher containsObjectAsJson(
+            Object expectedObject,
+            Class<T> targetClass) {
+        return mvcResult -> {
+            String json =mvcResult.getResponse().getContentAsString();
+            T actualObject = objectMapper.readValue(json, targetClass);
+            assertThat(actualObject).extracting("errors").isEqualToComparingFieldByField(expectedObject);
```

Comment:
```suggestion
        assertThat(actualObject)
                .usingRecursiveComparison()
                .isEqualTo(expectedObject);
```

`@Deprecated` 된 메소드는 동작은 되지만 사용 안하는게 좋습니다. 

저 메소드의 경우는 필드를 순회하는 동안 첫 번째 깊이의 필드만 비교가 되서 더 이상 사용하지 말라고 하네요

Reply:
```java
extracting("errors")
```
 `errors` 라는 필드가 없는 객체의 내부를 비교할 때 예외가 나옵니다

Reply:
`.usingRecursiveComparison()`
재귀적으로 돌면서 검증 할 때 쓰는 메서드군요..  좋은 정보 알아갑니다.

아 "errors" 부분 오타가 있었군요.. 테스트하면서 extracting을 썻는데 그대로 써버렸네요... 감사합니다.